lib/tag_parser.rb

```diff
@@ -0,0 +1,54 @@
+
+module TagParser
+
+
+  def self.parse_tag(string)
+
+    type = string.match /<(\w+)/
+    type = type[1] if type
+    tag = Tag.new(type)
+
+    string.scan /\s(\w+)\s*=\s*['|"](.+?)['|"]/ do |key, value|
+      case key
+      when 'class'
+        tag.classes= value.split(" ")
+      else
+        tag.send("#{key}=".to_sym, value)
+      end
+    end
+    tag
+  end
+end
+
+
+
+class Tag
+  attr_accessor :type, :classes, :id, :name, :depth, :children, :parent, :text
+
+  def initialize(type = nil)
+    @type = type
+    @children = []
+  end
+
+  def data_attributes
```

Comment:
Do you have a way to add attributes other than the ones listed here? I think you got most of them covered but I was just wondering.